plugins/async-node/jvm/src/test/kotlin/com/intuit/playerui/plugins/asyncnode/AsyncNodePluginTest.kt

```diff
@@ -0,0 +1,227 @@
+                "state_type": "END",
+                "outcome": "done",
+                "param": {
+                  "someKey": "someValue"
+                },
+                "extraKey": "extraValue",
+                "extraObject": {
+                  "someInt": 1
+                }
+              }
+            }
+          }
+        }
+        """.trimMargin()
+
+    override val plugins = listOf(AsyncNodePlugin())
+
+    private val plugin get() = player.asyncNodePlugin!!
+
+    @TestTemplate
+    fun `plugin is not null`() {
+        Assertions.assertNotNull(plugin)
+    }
+
+    @TestTemplate
+    fun `async node hook is tappable`() = runBlockingTest() {
+        var update: Asset? = null
+        plugin?.hooks?.onAsyncNode?.tap("") { _, node ->
+            BailResult.Bail(
+                listOf(
+                    mapOf(
+                        "asset" to mapOf(
+                            "id" to "asset-1",
+                            "type" to "text",
+                            "value" to "New asset!",
+                        ),
+                    ),
+                ),
+            )
+        }
+        var count = 0
+        suspendCancellableCoroutine { cont ->
+            player.hooks.view.tap { v ->
+                v?.hooks?.onUpdate?.tap { asset ->
+                    count++
+                    update = asset
+                    if (count == 2) cont.resume(true) {}
+                }
+            }
+            player.start(asyncNodeFlowSimple)
+        }
+        Assertions.assertTrue(count == 2)
+        Assertions.assertTrue((update?.get("actions") as List<*>).isNotEmpty())
+    }
+
+    @TestTemplate
+    fun `replace async node with multiNode`() = runBlockingTest {
+        var update: Asset? = null
+        plugin?.hooks?.onAsyncNode?.tap("") { _, node ->
+            BailResult.Bail(
+                listOf(
+                    mapOf(
+                        "asset" to mapOf(
+                            "id" to "asset-1",
+                            "type" to "text",
+                            "value" to "New asset!",
+                        ),
+                    ),
+                    mapOf(
+                        "asset" to mapOf(
+                            "id" to "asset-2",
+                            "type" to "text",
+                            "value" to "New asset!",
+                        ),
+                    ),
+                ),
+            )
+        }
+        var count = 0
+        suspendCancellableCoroutine { cont ->
+            player.hooks.view.tap { v ->
+                v?.hooks?.onUpdate?.tap { asset ->
+                    count++
+                    update = asset
+                    if (count == 2) cont.resume(true) {}
+                }
+            }
+            player.start(asyncNodeFlowSimple)
+        }
+        Assertions.assertTrue(count == 2)
+        Assertions.assertTrue((update?.get("actions") as List<*>).size == 3)
```

Comment:
Nit: could probably use `getList` helper along with `assertEquals`:
```suggestion
        Assertions.assertEquals(3, update?.getList("actions")?.size)
```

Reply:
done